TypeSystem/flow-analysis/type_of_interest_A03_t04.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that at type `T` is a type of interest for a variable `x`
+/// in a set of tested types `tested` if `tested` contains a type `S` such that
+/// `T` is `S`, or `T` is `NonNull(S)`.
+///
+/// @description Checks that a type `Null` cannot be made a type of interest
+/// for a variable.
```

Comment:
This one is more tricky. I actually think it should be possible to promote to `Null` (e.g., from `T?` for some `T`) when the construct that does it uses the type `Null` explicitly (so `s == null` won't do it).

When the variable has been promoted, we could use `expectStaticType<..>()` to verify that this  promotion has indeed occurred.